src/App.tsx

```diff
@@ -1,31 +1,155 @@
 import React from 'react';
 import './App.scss';
 import './styles/general.scss';
+import classNames from 'classnames';
 import { TodoList } from './components/TodoList';
 import { CurrentUser } from './components/CurrentUser';
+import { getTodos } from './api';
 
 interface State {
   selectedUserId: number;
+  todos: Todo[];
+  titleToSearch: string;
+  todosToShow: string;
+  loading: boolean;
 }
 
 class App extends React.Component<{}, State> {
   state: State = {
     selectedUserId: 0,
+    todos: [],
+    titleToSearch: '',
+    todosToShow: 'all',
+    loading: false,
+  };
+
+  async componentDidMount() {
+    const todos = await getTodos();
+
+    this.setState({ todos: [...todos] });
+  }
+
+  selectUser = (selectedUserId: number) => {
+    this.setState({ selectedUserId });
+    if (selectedUserId !== this.state.selectedUserId && selectedUserId !== 0) {
+      this.changeLoadingStatus();
+    }
+  };
+
+  prepareTodos = () => {
+    const {
+      todos,
+      titleToSearch,
+      todosToShow,
+    } = this.state;
+
+    let copiedTodos = [...todos];
+
+    if (titleToSearch) {
+      copiedTodos = copiedTodos.filter(
+        todo => todo.title.toLowerCase().includes(titleToSearch.toLowerCase()),
```

Comment:
create a separate variable for `titleToSearch.toLowerCase()`, no need to do it on every iteration